### PR TITLE
add messages for ir vision in look_around

### LINKED ISSUE #12339: look_around doesn't describe monsters seen with IR vision.
as title, highlighting the red '?' drawn for a monster seen with IR vision should describe the monster, at the least with a general, "You see a figure radiating heat".
## <bountysource-plugin>

Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/14142007-look_around-doesn-t-describe-monsters-seen-with-ir-vision?utm_campaign=plugin&utm_content=tracker%2F146201&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F146201&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by floyza**
#### Summary

Interface "Describe monsters seen by IR and other special types of vision when highlighted in look_around"

#### Purpose of change

Fixes #12339

#### Describe the solution

Adds a check into print_all_tile_info for sees_with_infrared and sees_with_specials

#### Testing
Spawn some creatures and add IR vision and electroreceptors mutations.
Look at creatures in darkness and through walls using look_around.

### CODE REVIEW

File: src/game.cpp [hunk 1]
```diff
@@ -5857,6 +5857,35 @@ void game::print_all_tile_info( const tripoint &lp, const catacurses::window &w_
         case VIS_HIDDEN:
             print_visibility_info( w_look, column, line, visibility );
 
+            if( creature != nullptr ) {
+                if( u.sees_with_infrared( *creature ) ) {
+                    std::string size_str;
+                    switch( creature->get_size() ) {
+                        case MS_TINY:
+                            size_str = "tiny";
```
Comment (Qrox) [inline src/game.cpp:5865]:
```suggestion
                            size_str = pgettext( "infrared size", "tiny" );
```

File: src/game.cpp [hunk 2]
```diff
@@ -5857,6 +5857,35 @@ void game::print_all_tile_info( const tripoint &lp, const catacurses::window &w_
...
+                            break;
+                        case MS_SMALL:
+                            size_str = "small";
```
Comment (Qrox) [inline src/game.cpp:5868]:
```suggestion
                            size_str = pgettext( "infrared size", "small" );
```

File: src/game.cpp [hunk 3]
```diff
@@ -5857,6 +5857,35 @@ void game::print_all_tile_info( const tripoint &lp, const catacurses::window &w_
...
+                            break;
+                        case MS_MEDIUM:
+                            break;
+                        case MS_LARGE:
+                            size_str = "large";
```
Comment (Qrox) [inline src/game.cpp:5873]:
```suggestion
                            size_str = pgettext( "infrared size", "large" );
```

File: src/game.cpp [hunk 4]
```diff
@@ -5857,6 +5857,35 @@ void game::print_all_tile_info( const tripoint &lp, const catacurses::window &w_
...
+                            break;
+                        case MS_HUGE:
+                            size_str = "huge";
```
Comment (Qrox) [inline src/game.cpp:5876]:
```suggestion
                            size_str = pgettext( "infrared size", "huge" );
```

File: src/game.cpp [hunk 5]
```diff
@@ -5857,6 +5857,35 @@ void game::print_all_tile_info( const tripoint &lp, const catacurses::window &w_
...
+                        case MS_MEDIUM:
+                            break;
+                        case MS_LARGE:
+                            size_str = pgettext( "infrared size", "large" );
+                            break;
+                        case MS_HUGE:
+                            size_str = pgettext( "infrared size", "huge" );
+                            break;
+                    }
+                    if( size_str != "" ) {
```
Comment (Night-Pryanik) [inline src/game.cpp:5879]:
```suggestion
                    if( !size_str.empty() "" ) {
```

File: src/game.cpp [hunk 6]
```diff
@@ -5857,6 +5857,35 @@ void game::print_all_tile_info( const tripoint &lp, const catacurses::window &w_
...
+                    std::string size_str;
+                    switch( creature->get_size() ) {
+                        case MS_TINY:
+                            size_str = pgettext( "infrared size", "tiny" );
+                            break;
+                        case MS_SMALL:
+                            size_str = pgettext( "infrared size", "small" );
+                            break;
+                        case MS_MEDIUM:
+                            break;
```
Comment (Night-Pryanik) [inline src/game.cpp:5871]: Nothing on medium size?

**Comment by ninjagamer76** [comment]
There's also a map of size_names in monster.cpp that could be moved into a header and used to replace the switch statement

**Pull request merged**